### PR TITLE
Capture native thread states

### PR DESCRIPTION
**Pull request by lemnik**
## Goal
Include a list of application threads and their statuses in crashes produced by the NDK plugin. The data reported includes:
- id
- name
- state

for each thread running at the time the report is generated. This reporting can be enabled or disabled using the existing [`sendThreads`](https://docs.bugsnag.com/platforms/android/configuration-options/#sendthreads) option in `Configuration`.

## Design
Scan the `/proc/self/task` directory to retrieve the list of running tasks / threads using signal-safe functions. The status of these threads is read (where possible) into a pre-allocated buffer of `bsg_thread`s from [`/proc/self/task/{tid}/stat`](https://man7.org/linux/man-pages/man5/proc.5.html) and mapped to human-readable values where required.

## Testing
New unit tests and Mazerunner scenarios to test that the data is reported when expected, and not reported when the configuration forbids it. Manual end-to-end testing has also been done to ensure the data is ingested and rendered as expected.

### CODE REVIEW

File: examples/sdk-app-example/app/src/main/java/com/example/bugsnag/android/ExampleApplication.kt [hunk 1]
```diff
@@ -26,7 +26,7 @@ class ExampleApplication : Application() {
         super.onCreate()
 
         val config = Configuration.load(this)
-        config.setUser("123456", "joebloggs@example.com", "Joe Bloggs")
+        config.setUser("123456", "joebloggs@example.com", "Joe Bzloggs")
```
Comment (fractalwrench) [inline examples/sdk-app-example/app/src/main/java/com/example/bugsnag/android/ExampleApplication.kt:29]: Just checking that this is intentional and not a stray key touch?

File: bugsnag-plugin-android-ndk/src/main/java/com/bugsnag/android/ndk/NativeBridge.kt [hunk 2]
```diff
@@ -172,9 +170,7 @@ class NativeBridge : StateObserver {
                     arg.autoDetectNdkCrashes,
                     Build.VERSION.SDK_INT,
                     is32bit,
-                    makeSafe(arg.appVersion ?: ""),
-                    makeSafe(arg.buildUuid ?: ""),
-                    makeSafe(arg.releaseStage ?: "")
```
Comment (fractalwrench) [inline bugsnag-plugin-android-ndk/src/main/java/com/bugsnag/android/ndk/NativeBridge.kt:177]: Are these definitely not required? IIRC there is a ticket in the backlog that should address whether these should be populated by native events

**Review by fractalwrench — APPROVED** [review]

File: bugsnag-plugin-android-ndk/src/main/java/com/bugsnag/android/ndk/NativeBridge.kt (hunk 2 above)
Comment (lemnik) [reply bugsnag-plugin-android-ndk/src/main/java/com/bugsnag/android/ndk/NativeBridge.kt:177]: They're not actually on the JNI signatures (in C) at the moment, so we definitely shouldn't be passing them down.

File: examples/sdk-app-example/app/src/main/java/com/example/bugsnag/android/ExampleApplication.kt (hunk 1 above)
Comment (lemnik) [reply examples/sdk-app-example/app/src/main/java/com/example/bugsnag/android/ExampleApplication.kt:29]: that was definitely a stray key

**Review by fractalwrench — APPROVED** [review]

File: bugsnag-plugin-android-ndk/src/main/jni/utils/threads.c [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+end:
+  // success if we hit the DONE marker
+  return state == PARSE_DONE;
+}
+
+/*
+ * Reads the /stat file fields we are looking for (TID, Name, State) into `dest`
+ * with a signal-safe approach.
+ */
+bool read_thread_state(bsg_thread *dest, const char *tid) {
```
Comment (kattrali) [inline bugsnag-plugin-android-ndk/src/main/jni/utils/threads.c:143]: The helper functions in this file have fairly generic names, which could cause a conflict somewhere. They should be marked static so they aren't visible elsewhere.
```suggestion
static bool read_thread_state(bsg_thread *dest, const char *tid) {
```

**Review by kattrali — APPROVED** [review]
a few procedural comments but looks good!

**Pull request merged**